symphony/bdk/core/client/api_client_factory.py

```diff
@@ -9,6 +13,25 @@
 from symphony.bdk.gen.api_client import ApiClient
 from symphony.bdk.gen.configuration import Configuration
 
+trace_id_context = ContextVar("trace_id_context")
```

Comment:
I would perhaps distinguish both, the trace id context that can be used everywhere in the bot, printed in the log 
and the logic to retrieve the trace id from the context, set it as a header to propagate it downstream